build-scripts/xc-cargo.sh

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+#
+# This is a small wrapper for running `cargo` inside of an XCode build,
+# which unfortunately doesn't seem to work quite right out-of-the-box.
+set -eEuvx
+
+
+# XCode tries to be helpful and overwrites the PATH. Reset that.
+PATH="$(bash -l -c 'echo $PATH')"
```

Comment:
This is what our existing `xc-universal-binary.sh` script does, and as noted [here](https://github.com/mozilla/application-services/pull/4052#issuecomment-851755959) the XCode build issues do seem to be related to `$PATH`..so 🤷🏻‍♂️